Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(VDir target)
+{	  
+    if(Handle_Ammo())
+    {
```

Comment:
Yes the make movement pattern do make segmentation fault(I think because I was trying to do without it and it worked fine(well it wasn't moving but was displayed on the map without crash).